apps/server/src/modules/files-storage/client/s3-client.adapter.spec.ts

```diff
@@ -46,131 +46,190 @@ describe('S3ClientAdapter', () => {
+					})
+				);
+			});
+
+			it('should return file', async () => {
+				setup();
+
+				const result = await service.get(pathToFile);
+
+				expect(result).toStrictEqual(
+					expect.objectContaining({
+						contentType: 'data.ContentType',
+						contentLength: 'data.ContentLength',
+						contentRange: 'data.ContentRange',
+						etag: 'data.ETag',
+					})
+				);
+			});
 		});
 
-		it('should call send() of client with bytes range', async () => {
-			await service.get(pathToFile, bytesRange);
+		describe('WHEN client throws error', () => {
+			const setup = () => {
+				const error = new Error('NoSuchKey');
+				// @ts-expect-error Testcase
+				client.send.mockRejectedValueOnce(error);
 
-			expect(client.send).toBeCalledWith(
-				expect.objectContaining({
-					input: { Bucket: config.bucket, Key: pathToFile, Range: bytesRange },
-				})
-			);
-		});
+				return { error };
+			};
 
-		it('should return file', async () => {
-			const result = await service.get(pathToFile);
-			expect(result).toStrictEqual(
-				expect.objectContaining({
-					contentType: 'data.ContentType',
-					contentLength: 'data.ContentLength',
-					contentRange: 'data.ContentRange',
-					etag: 'data.ETag',
-				})
-			);
+			it('should throw error', async () => {
+				const { error } = setup();
+
+				await expect(service.get(pathToFile)).rejects.toThrowError(error);
+			});
 		});
+	});
 
-		it('should throw error from s3 client if NoSuchKey', async () => {
-			const e = new Error('NoSuchKey');
-			client.send = jest.fn().mockRejectedValue(e);
+	describe('create', () => {
+		const createFileAndPath = () => {
+			const readable = Readable.from('ddd');
+			const file = new FileDto({
+				data: readable,
+				name: 'test.txt',
+				mimeType: 'text/plain',
+			});
+			const path = 'test/test.txt';
+
+			return { file, path };
+		};
 
-			await expect(service.get(pathToFile)).rejects.toThrowError(new NotFoundException('NoSuchKey'));
-		});
+		describe('WHEN file is created successfully', () => {
+			const setup = () => {
+				const { file, path } = createFileAndPath();
 
-		it('should throw error from client', async () => {
-			const e = new Error('Bad Request');
-			client.send = jest.fn().mockRejectedValue(e);
+				client.config.endpoint = jest.fn().mockResolvedValueOnce({ protocol: '' });
 
-			await expect(service.get(pathToFile)).rejects.toThrowError();
-		});
-	});
+				return { file, path };
+			};
 
-	describe('create', () => {
-		const readable = Readable.from('ddd');
-		const file = new FileDto({
-			data: readable,
-			name: 'test.txt',
-			mimeType: 'text/plain',
+			it('should return data', async () => {
+				const { file, path } = setup();
+
+				const result = await service.create(path, file);
+
+				expect(result).toBeDefined();
+			});
 		});
-		const path = 'test/test.txt';
 
-		it('should return data', async () => {
-			client.send = jest.fn().mockResolvedValue(true);
+		describe('WHEN client throws Bad Request error', () => {
```

Comment:
This is not restricted to Bad Request errors. You could move this test below the test with the NoSuchBucket error and name it maybe  'WHEN client throws any other error than NoSuchBucket'.